lib/ansible/modules/cloud/azure/azure_rm_image_facts.py

```diff
@@ -0,0 +1,187 @@
+except:
+    pass
+
+
+class AzureRMImageFacts(AzureRMModuleBase):
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            name=dict(
+                type='str',
+                required=False
+            ),
+            resource_group=dict(
+                type='str',
+                required=False
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.tags = None
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_images=None)
+        )
+
+        super(AzureRMImageFacts, self).__init__(
+            derived_arg_spec=self.module_arg_spec,
+            supports_check_mode=True,
+            supports_tags=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.results['ansible_facts'] = (
+            self.get_item() if (self.name or self.resource_group)
+            else self.list_items())
+
+        return self.results
+
+    def _list_images(self):
+        try:
+            images = self.compute_client.images
+            image_list = images.list()
+        except CloudError:
+            self.fail("No images found!")
+        except Exception as e:
+            self.fail("An exception occurred: {0}".format(str(e)))
+
+        named_images = []
+
+        for image in image_list:
+
+            data_disks = []
+
+            for disks in image.storage_profile.data_disks:
+                data_disks.append(dict(lun=disks.lun,
+                                       disk_size_gb=disks.disk_size_gb,
+                                       managed_disk=(not (disks.managed_disk is None))))
+
+            os_disk = dict(disk_size_gb=image.storage_profile.os_disk.disk_size_gb,
+                           managed_disk=(not (image.storage_profile.os_disk.managed_disk is None)))
+
+            image_info = dict(name=image.name,
+                              location=image.location,
+                              resource_group=image.id.split("/")[4],
+                              managed=(not (image.storage_profile.os_disk.managed_disk is None)),
+                              id=image.id,
+                              storage_profile=dict(os_disk=os_disk,
+                                                   data_disks=data_disks))
+            named_images.append(image_info)
+
+        return named_images
+
+    def list_items(self):
+
+        image_names = self._list_images()
+
+        exists = not (not image_names)
+
+        return dict(azure_images=image_names,
+                    exists=exists,
+                    changed=False)
+
+    def get_item(self):
+
+        image_names = self._list_images()
+        image_item = []
+
+        found = False
+
+        for image in image_names:
+            if self.name and self.resource_group:
```

Comment:
if the `self.name` is specified, here call the `list` API has a very heavy overhead. How about calling the `get(resource_group, name)` API?